src/Dialogs/Weather/EdlSettingsWidget.cpp

```
@@ -0,0 +1,251 @@
+      const unsigned selected_index =
+        SelectedCachedDayIndex(cached_days);
+      for (unsigned i = 0; i < cached_days.size(); ++i)
+        df.AddChoice(i, FormatCachedDayLabel(cached_days[i]).c_str());
+
+      df.SetValue(selected_index);
+    }
+
+    control.RefreshDisplay();
+  }
+
+  if (precache_day_button != nullptr)
```

Cppcheck (2.21.0)

[error] Dereference of an invalid iterator

(derefInvalidIterator)